example/apps/app/tests/__init__.py

```diff
@@ -201,3 +204,107 @@ def test_generic_m2m_manager_should_get_object_or_none_accroding_to_model_class(
         assert_equal(
             m2m_inst.related_objects.get_object_or_none(OneRelatedObject, related_object_inst3.pk), related_object_inst3
         )
+
+    def test_named_generic_m2m_should_add_related_object(self):
+        m2m_inst = NamedGenericManyToManyModel.objects.create()
+        related_object_inst1 = OneRelatedObject.objects.create()
+        related_object_inst2 = SecondRelatedObject.objects.create(id='unique')
+        related_object_inst3 = OneRelatedObject.objects.create()
+
+        assert_equal(m2m_inst.related_objects.count(), 0)
+        m2m_inst.related_objects.add(related_object1=related_object_inst1)
+        assert_equal(m2m_inst.related_objects.count(), 1)
+        assert_equal(m2m_inst.related_objects.related_object1, related_object_inst1)
+
+        m2m_inst.related_objects.add(related_object2=related_object_inst2, related_object3=related_object_inst3)
+        assert_equal(m2m_inst.related_objects.count(), 3)
+        assert_equal(m2m_inst.related_objects.related_object1, related_object_inst1)
+        assert_equal(m2m_inst.related_objects.related_object2, related_object_inst2)
+        assert_equal(m2m_inst.related_objects.related_object3, related_object_inst3)
+
+        # Duplicate named objects are updated
+        m2m_inst.related_objects.add(related_object1=related_object_inst2)
```

Comment:
I have to remember this, it's convenient but a bit unexpected.

Reply:
It makes sense for me, what should be expected behavior in your opinion?

Reply:
I would just expect a separate method for updating, this is more convenient, though

Reply:
For me this is better than separate function. The second reason is that I need to override default django behavior. Without it exception will be returned and this is what we don't wan.